sdk/core/core-rest-pipeline/review/core-rest-pipeline.api.md

```diff
@@ -398,6 +398,9 @@ export interface ThrottlingRetryPolicyOptions {
     maxRetries?: number;
 }
 
+// @public
+export function throttlingRetryStrategy(): RetryStrategy;
```

Comment:
I wonder if there's a way we can avoid exposing this function, like some kind of factory that you can provide additional strategies to and optionally disable default strategies. Something like

```ts
function extendDefaultRetryPolicy(additionalStrategies: RetryStrategy[], options?: { }): PipelinePolicy
```

thoughts?

Reply:
I like this idea, but I'm wondering if it would be better to extend `PipelineOptions.retryOptions` and allow the user to specify a `RetryStrategy[]` array that they can make with a builder/factory. 

I agree that users should be able to disable the default strategies, but I'm concerned about avoiding having our solution depend on the internals of `defaultRetryPolicy` - specifically that there's exactly two named strategies. Disabling them by name/index without exposing the actual strategies means that changing the structure of the default strategies (even without changing their behavior) could break a user's code. 

Maybe we limit the user to only inserting strategies between the throttling and exponential strategies? That way, instead of having to guarantee the names and positions don't change, we only need to guarantee the behavior before and after the injected strategies. 

Alternatively, the entire feature might be more useful in the pipeline package. I can imagine any of the SDKs could benefit from failover endpoints, and the new strategy can just go in between the throttling and exponential strategies as is (along with system errors as discussed above).

Reply:
I think I have something more palatable to serve this iteration. Open for comments!